src/main/java/htsjdk/samtools/CSIIndex.java

```diff
@@ -0,0 +1,466 @@
+package htsjdk.samtools;
+
+import htsjdk.samtools.seekablestream.SeekableStream;
+import htsjdk.samtools.util.RuntimeIOException;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.util.*;
+
+
+public class CSIIndex extends AbstractBAMFileIndex implements BrowseableBAMIndex {
+
+    private int binDepth;
```

Comment:
I'd love to make these final but I also want the methods that set them to not just be inlined into the constructor... I hate that java makes me choose.